src/SheetsRegistry.js

```diff
@@ -1,12 +1,22 @@
 /* @flow */
-import type {ToCssOptions} from './types'
+import {createGenerateClassName} from './utils/generateClassName'
+import type {ToCssOptions, SheetsRegistryOptions} from './types'
 import type StyleSheet from './StyleSheet'
 
 /**
  * Sheets registry to access them all at one place.
  */
 export default class SheetsRegistry {
   registry: Array<StyleSheet> = []
+  registryOptions: SheetsRegistryOptions
+
+  constructor(options: SheetsRegistryOptions = {}) {
+    const generateClassName = options.generateClassName ||
+      (options.resetClassNameCounter === true && createGenerateClassName())
```

Comment:
why do you need an option? Why not to reset the counter every time a new registry is created?